### PR TITLE
Adds teeth. Adds ability to pull out teeth. Adds ability to put teeth in a glass. Adds ability to eat teeth.

### PR DESCRIPTION
**Pull request by drexample**
Adds teeth, ability to pull them out, put them in a glass and eat them.
A must have feature.

...Into your mouth, rapturously, one by one. They go down troublesomely, but down they go. "Who's there?" the Bucculent Dentist calls. Who indeed? An excellent question. You fade back into the street, stomach rattling, mouth awash with murmurous red. 

All shall be well.
And all shall be well.
And all manner of things shall be well.
default

You can reattach them through surgery.
But why?

### CODE REVIEW

**Comment by Iroquois-Pliskin** [comment]
Full support.

**Comment by Wickedtemp** [comment]
I'd rather not have body-horror type stuff.

**Comment by cruxxansata** [comment]
> 
> 
> I'd rather not have body-horror type stuff.

What about Brutal Predation? That's way more graphic. That's not even the most graphic thing vorecode has to offer.

**Comment by Sebbe9123** [comment]
Can't wait for this to be abused heavily and gross me the fuck out.

**Comment by Screemonster** [comment]
w h a t

**Comment by drexample** [comment]
> 
> 
> Can't wait for this to be abused heavily and gross me the fuck out.

Pulling out teeth hurts their owner.
Eating teeth hurts the eater.
The price we have to pay.

**Comment by Wickedtemp** [comment]
If its covered by the same prefbreak rules as brutal predation, sure.

File: code/modules/mob/living/carbon/human/human_defines_vr.dm [hunk 1]
```diff
@@ -7,3 +7,4 @@
 	var/flapping = 0
 	var/vantag_pref = VANTAG_NONE //What's my status?
 	var/impersonate_bodytype //For impersonating a bodytype
+	var/teeth = 32 //How many teeth are left in your mouth
```
Comment (Screemonster) [inline code/modules/mob/living/carbon/human/human_defines_vr.dm]: Unacceptable, not all species on the station have 32 teeth. Some may not have teeth at all, some might have chitinous mandibles or somesuch that superficially resemble the function of teeth but are not themselves teeth

**Comment by Iroquois-Pliskin** [comment]
I hope you intend to let Promethean implant teeth.

Cheri needs an award winning smile.

**Comment by Screemonster** [comment]
now I'm just picturing cheri getting into barfights so she can knock people's teeth out and steal them for herself

File: code/modules/mob/living/carbon/human/human_defines_vr.dm (hunk 1 above)
Comment (drexample) [reply code/modules/mob/living/carbon/human/human_defines_vr.dm]: For the species, which do not posses teeth, I have added a variable in species defines.
As for my mistake of assuming that all species have equal amount of teeth - I admit that oversight. Where, or from whom can I get the exact numbers. I do wish to correct my mistake so.

**Comment by drexample** [comment]
> 
> 
> I hope you intend to let Promethean implant teeth.
> 
> Cheri needs an award winning smile.

Of course. The joy of dental possession must be shared.
Prometheans are capable of being implanted teeth via surgery.

**Comment by drexample** [comment]
> 
> 
> Since you compared it to brutal predation, if this is covered by the same prefbreak rules as brutal predation, sure, I'd be okay with it.
> 
> Teeth stuff - pulling out teeth, teeth falling out, its something I, and quite a few others, are squeamish about. You ever had those dreams where your teeth just fall out? Those kinds of dreams fuck with me on a deeper level than nearly anything else.
> 
> I really don't want to see that shit in the game I play more than any other. At the absolute minimum if this even gets accepted --which I still kinda personally hope it doesn't because it seems more like a "Hey lets see if we can get this ingame it'll be so funny if it does!" thing more than an actually desired feature-- it should go under the same pref rules as every other removed part of the body.

I seem to not share your sense of humor, as I do not find removal or consumption of teeth humorous. As for interaction rules and preferences - they are dictated by the higher order of administration. I'm not the one, who writes rules, I'm the one who provides the means and possibilities. May all be well for you and be as you desire.

**Comment by Iroquois-Pliskin** [comment]
Y E S

File: code/game/objects/items/stacks/teeth.dm [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+/obj/item/teethglass
+	name = "glass of teeth"
+	desc = "You get hungry by just looking at them."
+	icon = 'icons/obj/teeth.dmi'
+	icon_state = "teethglass"
+
+
+/obj/item/teethglass/attack(mob/M as mob, mob/user as mob)
+	if (!ishuman(M))
+		..()
```
Comment (kevinz000) [inline code/game/objects/items/stacks/teeth.dm]: return ..()

File: code/game/objects/items/stacks/teeth.dm [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+		return
+	else
+		var/mob/living/carbon/human/H = M
+		var/obj/item/blocked = H.check_mouth_coverage()
+		if(blocked)
+			to_chat(user, "<span class='warning'>\The [blocked] is in the way!</span>")
+			return
+		if(H.species.has_organ[O_LIVER])
+			if(H == user)
+				user.visible_message("[user] has swallowed a glass of teeth!", "You swallow the teeth!")
```
Comment (kevinz000) [inline code/game/objects/items/stacks/teeth.dm]: swallows*

File: code/game/objects/items/stacks/teeth.dm [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+					to_chat(user, "<span class='warning'>WHY ARE YOU SO DAMNABLY HUNGRY?!</span>")
+				L.take_damage(40)
+			else
+				user.visible_message("[user] tries to feed [M] a glass of teeth!")
+				if(do_after(user, 20))
+					user.visible_message("[user] has fed [M] a glass of teeth!", "You feed [M] the teeth!")
+					var/obj/item/organ/internal/liver/L = H.internal_organs_by_name[O_LIVER]
+					L.take_damage(50)
+		playsound(src.loc, 'sound/vore/gulp.ogg', 25, 1)
+		new /obj/item/weapon/reagent_containers/food/drinks/drinkingglass(user.loc)
```
Comment (kevinz000) [inline code/game/objects/items/stacks/teeth.dm]: drop_location() i believe?

File: code/game/objects/items/stacks/teeth.dm [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+					H.verbs |= /mob/living/proc/eat_trash
+					to_chat(user, "<span class='warning'>WHY ARE YOU SO DAMNABLY HUNGRY?!</span>")
+				L.take_damage(40)
+			else
+				user.visible_message("[user] tries to feed [M] a glass of teeth!")
+				if(do_after(user, 20))
+					user.visible_message("[user] has fed [M] a glass of teeth!", "You feed [M] the teeth!")
+					var/obj/item/organ/internal/liver/L = H.internal_organs_by_name[O_LIVER]
+					L.take_damage(50)
+		playsound(src.loc, 'sound/vore/gulp.ogg', 25, 1)
```
Comment (kevinz000) [inline code/game/objects/items/stacks/teeth.dm]: src

File: code/game/objects/items/stacks/teeth.dm [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+		if(blocked)
+			to_chat(user, "<span class='warning'>\The [blocked] is in the way!</span>")
+			return
+		if(H.species.has_organ[O_LIVER])
+			if(H == user)
+				user.visible_message("[user] has swallowed a handfull of teeth!", "You swallow the teeth!")
+				H.nutrition -= amount*3 //NOT ENOUGH NOT ENOUGH NOT ENOUGH
+				var/obj/item/organ/internal/liver/L = H.internal_organs_by_name[O_LIVER]
+				L.take_damage(amount*1.5)
+				playsound(src.loc, 'sound/vore/gulp.ogg', 25, 1)
```
Comment (kevinz000) [inline code/game/objects/items/stacks/teeth.dm]: src

File: code/game/objects/items/stacks/teeth.dm [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+			else
+				user.visible_message("[user] tries to feed [M] a handfull of teeth!")
+				if(do_after(user, 20))
+					user.visible_message("[user] has fed [M] a handfull of teeth!", "You feed [M] the teeth!")
+					var/obj/item/organ/internal/liver/L = H.internal_organs_by_name[O_LIVER]
+					L.take_damage(amount*1.5)
+					playsound(src.loc, 'sound/vore/gulp.ogg', 25, 1)
```
Comment (kevinz000) [inline code/game/objects/items/stacks/teeth.dm]: src

File: code/game/objects/items/stacks/teeth.dm [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+	force = 2.0
+	throwforce = 7.0
+	throw_speed = 5
+	throw_range = 20
+	max_amount = 32
+
+
+/obj/item/stack/teeth/attack(mob/M as mob, mob/user as mob)
+	if (!ishuman(M))
+		..()
```
Comment (kevinz000) [inline code/game/objects/items/stacks/teeth.dm]: return ..()

File: code/game/objects/items/stacks/teeth.dm [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+		return
+	else
+		var/mob/living/carbon/human/H = M
+		var/obj/item/blocked = H.check_mouth_coverage()
+		if(blocked)
+			to_chat(user, "<span class='warning'>\The [blocked] is in the way!</span>")
+			return
+		if(H.species.has_organ[O_LIVER])
+			if(H == user)
+				user.visible_message("[user] has swallowed a handfull of teeth!", "You swallow the teeth!")
```
Comment (kevinz000) [inline code/game/objects/items/stacks/teeth.dm]: swallows

File: code/modules/food/food/drinks/drinkingglass.dm [hunk 10]
```diff
@@ -52,6 +52,18 @@
 			center_of_mass = list("x"=16, "y"=10)
 			return
 
+	attackby(obj/item/I, mob/user)
```
Comment (kevinz000) [inline code/modules/food/food/drinks/drinkingglass.dm]: should probably use absolute pathing

File: code/modules/food/food/drinks/drinkingglass.dm [hunk 11]
```diff
@@ -52,6 +52,18 @@
...
+		..()
```
Comment (kevinz000) [inline code/modules/food/food/drinks/drinkingglass.dm]: . = ..()?

File: code/modules/food/food/drinks/drinkingglass.dm [hunk 12]
```diff
@@ -52,6 +52,18 @@
...
+		if(istype(I, /obj/item/stack/teeth))
+			var/obj/item/stack/teeth/T = I
+			if (reagents.reagent_list.len == 0)
```
Comment (kevinz000) [inline code/modules/food/food/drinks/drinkingglass.dm]: !length(reagents.reagent_list) or !reagents.reagent_list.len

File: code/modules/food/food/drinks/drinkingglass.dm [hunk 13]
```diff
@@ -52,6 +52,18 @@
...
+				if(T.amount > 19)
```
Comment (kevinz000) [inline code/modules/food/food/drinks/drinkingglass.dm]: nitpick but would be neater to have it >= 20

File: code/modules/food/food/drinks/drinkingglass.dm [hunk 14]
```diff
@@ -52,6 +52,18 @@
...
+					new /obj/item/teethglass(src.loc)
+					T.amount -= 20
```
Comment (kevinz000) [inline code/modules/food/food/drinks/drinkingglass.dm]: T.use(20) or whatever the stack use proc is

File: code/modules/mob/living/carbon/human/species/species_vr.dm [hunk 15]
```diff
@@ -17,6 +17,8 @@
 	var/wing_animation
 	var/icobase_wing
 
+	var/hasteeth = 1 //If the species has teeth. Pity those who don't. They can still take them from those, who do, though.
```
Comment (kevinz000) [inline code/modules/mob/living/carbon/human/species/species_vr.dm]: TRUE

File: code/modules/mob/living/carbon/human/species/station/station_vr.dm [hunk 16]
```diff
@@ -123,6 +123,8 @@
 	color_mult = 1
 	inherent_verbs = list(/mob/living/proc/shred_limb,/mob/living/proc/flying_toggle,/mob/living/proc/start_wings_hovering)
 
+	hasteeth = 0
```
Comment (kevinz000) [inline code/modules/mob/living/carbon/human/species/station/station_vr.dm]: FALSE

File: code/modules/mob/living/carbon/human/species/station/station_vr.dm [hunk 17]
```diff
@@ -332,6 +334,9 @@
 
 /datum/species/diona
 	spawn_flags = SPECIES_CAN_JOIN | SPECIES_IS_WHITELISTED
+
+	hasteeth = 0
```
Comment (kevinz000) [inline code/modules/mob/living/carbon/human/species/station/station_vr.dm]: FALSE

File: code/modules/mob/living/carbon/human/species/station/station_vr.dm [hunk 18]
```diff
@@ -352,6 +357,7 @@
 		)
 
 /datum/species/shapeshifter/promethean
+	hasteeth = 0
```
Comment (kevinz000) [inline code/modules/mob/living/carbon/human/species/station/station_vr.dm]: FALSE

File: code/modules/mob/living/carbon/human/species/station/station_vr.dm [hunk 19]
```diff
@@ -363,6 +369,7 @@
 	base_color = "#EECEB3"
 
 /datum/species/vox
+	hasteeth = 0
```
Comment (kevinz000) [inline code/modules/mob/living/carbon/human/species/station/station_vr.dm]: FALSE

File: code/modules/surgery/face.dm [hunk 20]
```diff
@@ -154,4 +154,40 @@
...
+	min_duration = 20
+	max_duration = 35
+
+/datum/surgery_step/face/fix_teeth/can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+	return ..() && target.teeth < 33
+
+/datum/surgery_step/face/fix_teeth/begin_step(mob/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+	user.visible_message("[user] reattaches [target]'s tooth.", \
+	"You start reattaching [target]'s tooth.")
+	..()
```
Comment (kevinz000) [inline code/modules/surgery/face.dm]: return ..() I believe?

File: code/modules/surgery/face.dm [hunk 21]
```diff
@@ -154,4 +154,40 @@
...
+	user.visible_message("<font color='blue'>[user] reattached [target]'s tooth back in place.</font>",	\
+	"<font color='blue'>You have reattached [target]'s tooth back in place.</font>")
+	var/obj/item/stack/teeth/T = tool
+	target.teeth += 1
+	T.use(1)
+
+
+/datum/surgery_step/face/fix_teeth/fail_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+	var/obj/item/organ/external/affected = target.get_organ(target_zone)
+	user.visible_message("<font color='red'>[user]'s hand slips, tearing skin in [target]'s mouth with \the [tool]!</font>", \
```
Comment (kevinz000) [inline code/modules/surgery/face.dm]: <span class='warning'>

File: code/modules/surgery/face.dm [hunk 22]
```diff
@@ -154,4 +154,40 @@
...
+	"<font color='red'>Your hand slips, tearing skin on [target]'s face with \the [tool]!</font>")
+	target.apply_damage(10, BRUTE, affected, sharp=1, sharp=1)
```
Comment (kevinz000) [inline code/modules/surgery/face.dm]: are you sure there's two arguments named sharp? also shouldn't it be TRUE?

File: code/modules/surgery/face.dm (hunk 20 above)
Comment (drexample) [reply code/modules/surgery/face.dm]: Copied from another surgery operation.
default

File: code/modules/surgery/face.dm (hunk 22 above)
Comment (drexample) [reply code/modules/surgery/face.dm]: default
 Copied from another surgery step.

**Comment by drexample** [comment]
> 
> 
> Hi, I am nitpicking.

Much obliged for your assistance.

**Comment by MrSign** [comment]
I'm scared, frightened and I feel like this has the high-chance of pref-break for perhaps... EVERYONE ON THE SERVER! AAAAA!

**Comment by drexample** [comment]
> 
> 
> I'm scared, frightened and I feel like this has the high-chance of pref-break for perhaps... EVERYONE ON THE SERVER! AAAAA!

Do not fret. Actions, proposed in such Pull Request do take considerable amount of time and require the recipient to be absolutely still. 
Furthermore, such actions, when unconditional, unprovoked or done despite the established set of rules, are ought to be punished by community order enforcement.

**Comment by Sebbe9123** [comment]
> > I'm scared, frightened and I feel like this has the high-chance of pref-break for perhaps... EVERYONE ON THE SERVER! AAAAA!
> 
> Do not fret. Actions, proposed in such Pull Request do take considerable amount of time and require the recipient to be absolutely still.
> Furthermore, such actions, when unconditional, unprovoked or done despite the established set of rules, are ought to be punished by community order enforcement.

So what you're saying is that we should chalk it up to "no sexxy prs" and shut it down?

**Comment by drexample** [comment]
> 
> 
> > > I'm scared, frightened and I feel like this has the high-chance of pref-break for perhaps... EVERYONE ON THE SERVER! AAAAA!
> > 
> > 
> > Do not fret. Actions, proposed in such Pull Request do take considerable amount of time and require the recipient to be absolutely still.
> > Furthermore, such actions, when unconditional, unprovoked or done despite the established set of rules, are ought to be punished by community order enforcement.
> 
> So what you're saying is that we should chalk it up to "no sexxy prs" and shut it down?

I do not see a correlation between the contents of this PR and anything sexual. If anyone is willingly harming someone else's or their own health, using provided mechanics, it is violation of the rules. I do not propose such, neither I propose any crewman on crewman violence. I am only providing the means for more rich interactions for people interested. 
Your implications are preposterous.

**Comment by SpadesNeil** [comment]
Poe's Law is in effect.

**Comment by SpadesNeil** [comment]
Now that the feature freeze is over,
no

**Comment by drexample** [comment]
Cowards.

**Pull request closed without merging**